### ISSUE TITLE
Add a federated and local timeline

### ISSUE THREAD

**Issue by glottis0q** [mastodon/mastodon-ios#369]
- [ ] Federated timeline
- [ ] Local timeline

**Comment by braydofficial** [comment]
Yes, highly needed. Especially for smaller instances. However, I'd suggest making this opt-in in the settings, so people don't get confused about the different timelines that much, when they first join Mastodon.

**Comment by glottis0q** [comment]
> Yes, highly needed. Especially for smaller instances. However, I'd suggest making this opt-in in the settings, so people don't get confused about the different timelines that much, when they first join Mastodon. 

There could be a tutorial a few days after login if the user is new to mastodon idk

**Comment by FediVideos** [comment]
The local timeline is especially because it's a central part of the community on smaller instances.

I just saw the admin of one very long running respected smaller instance consider withdrawing entirely from the joinmastodon list because of the lack of local timeline in the app.

Mastodon really shouldn't be losing good small well-run instances, the local timeline should be brought back or at least somehow included as an option.

Federated is maybe less needed as it does tend to be a firehose most of the time, but Local can be really, really good if you're on a small or specialised instance.

**Comment by braydofficial** [comment]
> The local timeline is especially important because it's a central part of the community on smaller instances.
> 
> I just saw the admin of one very long running respected smaller instance consider withdrawing entirely from the joinmastodon list because of the lack of local timeline in the app.
> 
> Mastodon really shouldn't be losing good small well-run instances, the local timeline should be included in the app at least as an option.
> 
> Federated is maybe less needed as it does tend to be a firehose most of the time, but Local can be really, really good if you're on a small or specialised instance.

Agreed. I think you should be able to use the local and federated timeline - as said - as an opt-in in the settings. The federated timeline could just be disabled by the instance's admins, because I believe it will be their job to ensure moderation on their own instance. I've seen fairly big instances with a well moderated federated timeline so it can be possible. I believe it should be the users choice / the admins choice.

**Comment by FediVideos** [comment]
The instance I was talking about has now withdrawn from joinmastodon, this is very sad :(

**Comment by nmalcolm** [comment]
I run my own Mastodon instance. It's small, but my community enjoys interacting with each other. Without the local timeline it completely kills the ability to discover what other people are talking about. People who use the official apps will post "Hello?" and get replies from people who use third party apps or the web app but they're completely cut off from being able to do the same to others. They don't understand why and I have to tell them to not use the official apps.

The local timeline is also a way I use to moderate content being posted. I use the official app for posting and notifications, but if I want to check everything's good I have to switch to another app. This makes no sense.

The official apps are very similar to every other third party app: inconsistent with the web app, missing core features, what's the point of an official app if third party ones are better? Yes, I don't have to use it, but it's confusing for new users. I don't have any control over what app they use, and being told to install a different one is not a good user experience.

Please, listen to the community. Listen to the instance admins. Add the timelines to the official apps.

**Comment by FediVideos** [comment]
Not sure how to refer to this reply from here, but seems like it is relevant in this thread too:

> 
> On large instances, the risks of a Local timeline outweigh the benefits. On large instances, the Local timeline is a firehose just like Federated, impossible to read and full of potentially nasty content but impossible to browse for fun. Seems like a no-brainer to zap Local.
> 
> On small instances, it's the other way around. The benefits of a Local timeline outweigh the risks, it's not a firehose, it helps build the community and people are much less likely to see nasty content because the moderators have better control over troublemakers. Removing Local seems like a bad idea.
> 
> I'm not sure how this situation can be resolved, maybe Local could be for instances below a certain size?

(I posted this on the Android app's equivalent issue)

**Comment by Hugo-Trentesaux** [comment]
Duplicate of #221

**Comment by glottis0q** [comment]
> Duplicate of #221 

But it's closed. As you can see, many want to have this feature. At least in smaller servers (with less than 20k users @Gargron :) people tent to use it a lot as they (there are a lot of other reasons too, but most likely) chose the server for a specific topic.

**Comment by braydofficial** [comment]
Agree. Especially as it doesn't make sense IMO to don't add it to the iOS app while all timelines seems to be supported on the Android App. It should be continuous above both apps.

**Comment by glottis0q** [comment]
Awesome! Thanks @MainasuK ! 😍

**Comment by paspro** [comment]
Any chance to see a Federated Timeline added?

**Comment by JustYourBud** [comment]
> Any chance to see a Federated Timeline added?

I would like to see this too, I think it's almost a necessary addition at this point with the influx of new users. [This toot](https://mastodon.lol/@nathan/109332204578263723) really puts into perspective how confusing it might be for a new user joining through the mobile app--the whole of the fediverse is wildly inaccessible to them vs other apps, which is a problem when the concept of the fediverse is more than likely very new to most people on this platform in the first place. This wouldn't be a good first impression for them. I saw one person compare Mastodon to a mix of Twitter and Discord, and I think part of that stems from the fact that the app has no access to the federated timeline.

It seems like such a wild thing to not include when so many people are trying to show how much promise the fediverse can have for the mainstream, and this app is possibly the most mainstream way to access it right now. Even if it's an opt-in feature with a pop-up when first using the app (because I completely understand the downsides to having the timeline, it can be very chaotic), I think some form of access to it would be beneficial in the end. Personally, I actually haven't had any issues with browsing the federated timeline, but I understand that isn't the case for everyone.

**Comment by nourkagha** [comment]
I agree that local and federated tabs should be added at the bottom.

The current design is just confusing new users, especially when it's not a reflection of the web UI at all.

The decision to not include those timelines was to simplify things and not overwhelm new users, but everyone understands local and federated timelines.

It's awkward when you explain Mastodon and the different timelines to people and they join and can't find them, leaving them even more confused.

**Comment by joshuahiggins** [comment]
Just chiming in to say that a Federated timeline is a must. 

Also I feel that the way Community is hidden in the search / featured area is confusing. I feel that the Home view should have a button to switch between timeline types. This is a common pattern that users are already familiar with for switching between algo driven and time based timelines on social sites. Personally, when I click the settings button on the home view, that’s what I expect it to offer.

My suggestion would be replacing the settings button on the home view with a timeline switching button to swap between all 3 timeline views. The settings are already present on the profile tab anyway, which is a common place users would already expect.

**Comment by codybartfast** [comment]
As a new user I find the absence of Local and Federated to be confusing and frustrating.

I assumed parity with the Web version and have spent an age looking for how to get to a Local / Federated view, and I had no success finding anything on Google to tell me how to find them (or telling my they can't be found).  It's only because I'm a programmer that I eventually thought to check GitHub issues,  otherwise I'd still be trying to work out what I was doing wrong.

I won't claim to understand Mastodon's style nor how it _should_ be used, but I'm looking for something like a firehose so I can find some users and tags to follow.  Even if my home page were not tumbleweed, I imagine being able to periodically access a firehose would be valuable for finding something different.

(And thanks to all those who helped create such a classy app)

**Comment by parkertried** [comment]
I’m using a small instance and agree that federated timeline support should be added to the app. I’ve just resorted to 3rd party apps atm.

**Comment by pablo03v** [comment]
> I’m using a small instance

If you are referring to your instances timeline, you can access that via the "Community"-Tab. Otherwise I still think a federated timeline should be placed somewhere in the app.

@parkerhindman

**Comment by parkertried** [comment]
@pabloscloud I use a single user instance so the local timeline is useless for me, I was referring the federated timeline for assistance  discovering other users without being forced to just use hashtags.
